src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/SmartContent.js

```diff
@@ -52,11 +65,27 @@ export default class SmartContent extends React.Component<Props> {
             formInspector.resourceKey === provider ? formInspector.id : undefined
         );
 
-        autorun(this.handleFilterCriteriaChange);
+        smartContentStorePool.add(this.smartContentStore, excludeDuplicates);
+
+        this.filterCriteriaChangeDisposer = autorun(this.handleFilterCriteriaChange);
+
+        if (!excludeDuplicates || this.previousSmartContentStores.length === 0) {
+            this.smartContentStore.start();
+        } else {
+            when(
```

Comment:
is this `when()` necessary here? i think the `updateRecursiveExcludedIds` method of the `smartContentSotrePool` will already handle this

Reply:
Yeah, it is necessary, because otherwise there will be too many requests. We have to wait until the previous smart content is completely loaded before loading the next one excluding duplicates, because otherwise the request will be done first without the excluded ids, and as soon as the previous stores are loaded it will be done again with the excluded ids. The first request would be useless.

Reply:
this is really hard to understand. i guess its even harder if you dont know about the intention.. can we at least add some comment(s) why this is necessary?

Reply:
I'll add a comment